mock/storage_reads.go

```diff
@@ -185,3 +220,49 @@ func (c *GroupCursor) Err() error {
 func (c *GroupCursor) Stats() cursors.CursorStats {
 	return c.StatsFunc()
 }
+
+type StoreReader struct {
+	ReadFilterFn func(ctx context.Context, req *datatypes.ReadFilterRequest) (reads.ResultSet, error)
```

Comment:
nit, you use `func` as an abbreviation for "function" earlier in the file, would be good to choose one or the other.  I like `fn` myself.